### DISCUSSION TITLE
CodeCarbon

### DISCUSSION THREAD

**Discussion by stumpyfr** [Green-Software-Foundation/carbon-aware-sdk#81] [General]
Hi,

Niels Freier from BCG here.

Started a long time ago, with other key players, we created  [CodeCarbon](https://github.com/mlco2/codecarbon), clearly in the same mindset as this SDK, except it was Python oriented (for now).
We started before the GSF's standard but I am confident we can either retrieve a lot of knowledge/experience or why not "merge" the effort.

Curious to know if some of you already had a look at CodeCarbon?
Then let's see all together what are our options, I really like the approach API based on this SDK!

**Comment by Willmish** [comment]
Hi @stumpyfr ! 
First of all, nice to meet you :D. I wrote down some of my thoughts on this below, but if you would like to have a more detailed chat, do drop in on our weekly project syncs (they happen on zoom, every Tuesday at 8am BST - link + calendar invite can be found in pinned messages on [slack channel #proj-carbon-aware-sdk](https://greensoftwarefdn.slack.com/archives/C02JRAV4QEP/p1655720603214979))

This project looks super cool. I haven't seen it before, but maybe @vaughanknight might have? At a first glance, the scope of it seems to be a bit wider than the SDK's, as it seems to be estimating full carbon emissions of a running software - need to look more into your methodology for calculating it.

The SDK focuses on getting the **I** (Location-based marginal carbon emissions) parameter of the Software Carbon Intensity (SCI) score - more on GSF's current methodology can be found [here](https://github.com/Green-Software-Foundation/software_carbon_intensity/blob/dev/Software_Carbon_Intensity/Software_Carbon_Intensity_Specification.md#methodology-summary). The Carbon Aware SDK's scope will probably increase in the future, but that might be a continuing project **building on** the SDK. I might however be wrong, and the SDK has more in common than I think, @vaughanknight can probably give his insight on this.

What are your ideas on merging the efforts? Sharing/standarising the methodology for calculating the carbon emissions? 

What I definitely see as a possibility is including the SDK (when it gets released) in CodeCarbon as a source for location-based carbon intensity rates. At the moment it looks like CodeCarbon is using static data for this: https://github.com/mlco2/codecarbon/blob/master/codecarbon/data/cloud/impact.csv so making it dynamic using SDK outputs will probably give a more accurate carbon emissions estimate. Let me know if that is what you had in mind.

**Comment by stumpyfr** [comment]
Honestly no clear idea yet, just the motivation to not duplicate the efforts.

Your idea of using the SDK to be more dynamic for the `I` makes sense!, we can clearly imagine combining them, as an optional, yet, more accurate data source into CodeCarbon.

I will try to connect to one of the next project syncs to have this type of discussion live, always easier.

**Reply by vaughanknight** [reply]
Just had a chat with Szymon on this.  We had stumbled on it a while ago while we were still figuring out where our focus was going to be.  And yes, the main thing is for the Carbon Aware SDK to retrieve dynamic values of I across a multitude of sources.  Happy to chat further!

**Reply by stumpyfr** [reply]
ok, I guess the name made me think about something else.
but clearly, the first step as you mentioned!
